test/e2e/publish.e2e.js

```diff
@@ -75,9 +79,45 @@ describe('Publish Commands', () => {
 		const args = ['publish', eventName, '--public'];
 		const { stdout, stderr, exitCode } = await cli.run(args);
 
-		expect(stdout).to.equal(`Published public event: ${eventName}${'\n'}`);
+		expect(stdout).to.include(`Published public event: ${eventName}${os.EOL}`);
 		expect(stderr).to.equal('');
 		expect(exitCode).to.equal(0);
 	});
+
+	it('Publishes a product event', async () => {
+		const args = ['publish', eventName, '--product', PRODUCT_01_ID];
+		const { stdout, stderr, exitCode } = await cli.run(args);
+
+		expect(stdout).to.include(`Published private event: ${eventName} to product: ${PRODUCT_01_ID}${os.EOL}`);
+		expect(stderr).to.equal('');
+		expect(exitCode).to.equal(0);
+	});
+
+	it('Publishes a private product event', async () => {
+		const args = ['publish', eventName, '--product', PRODUCT_01_ID, '--private'];
+		const { stdout, stderr, exitCode } = await cli.run(args);
+
+		expect(stdout).to.include(`Published private event: ${eventName} to product: ${PRODUCT_01_ID}${os.EOL}`);
+		expect(stderr).to.equal('');
+		expect(exitCode).to.equal(0);
+	});
+
+	it('Publishes a public product event', async () => {
```

Comment:
[api docs](https://docs.particle.io/reference/device-cloud/api/#publish-a-product-event) say you can have `public` and `private` product events but i'll confess i'm foggy on how they work exactly

Reply:
There is no public product stream. The isPrivate flag is ignored for products. You can delete this test.

Reply:
@monkbroc per https://github.com/particle-iot/particle-cli/pull/559#discussion_r398244953 - see above. sounds like we need to update docs as well. i can take care of that :+1:

Reply:
Yeah that is inaccurate